### PR TITLE
Django and Pipenv

### PR DESCRIPTION
**Pull request by LiorSilberman**

### CODE REVIEW

**Comment by pazMenachem** [comment]
seems good to me.

**Comment by yftacherzog** [comment]
@LiorSilberman the buyTogether directory is essentially a python package. The convention for package names in python is all-lowercase. I suggest calling it buytogether or buy_together.
See some details here: https://peps.python.org/pep-0008/#package-and-module-names

**Comment by yftacherzog** [comment]
@LiorSilberman please use a more descriptive PR name.

**Comment by LiorSilberman** [comment]
@yftacherzog
1. If I want to rename the directory, do I need to add new commit for this?
2. What do I need to consider about? because Karam used this name in his files so he need to change the name in his files too?

**Comment by pazMenachem** [comment]
1. i assume its like modifying any other files, once youve changed the files you commit them and the same PR will change, thou you need to squash the commits like we did in challange#2, rebase -i Head~ command.
2. i think he needs to modify the files as well.
any mentor correct me if im wrong please.

**Comment by yftacherzog** [comment]
> 1. If I want to rename the directory, do I need to push new PR for this?

Your PR wants to introduced that directory, and the PR was not merged yet, so you need the PR content, rather than to push a new PR (or new commit).
You need to:
1. Make your modifications. 
2. use `git status` to see the unstaged changes.
3. You'll see the old files removed and the new ones added.
4. Use `git add` to add the changes for both removed and added files.
5. use `git commit --amend` to amend your last commit.
6. use `git push` with the `--force` option to overwrite your remote branch with your local changes.

**Comment by yftacherzog** [comment]
> 1. i assume its like modifying any other files, once youve changed the files you commit them and the same PR will change, thou you need to squash the commits like we did in challange#2, rebase -i Head~ command.

You can avoid rebasing or squashing in most cases (and this one as well) if you use `commit --amend`.

**Comment by yftacherzog** [comment]
> 2. What do I need to consider about? because Karam used this name in his files so he need to change the name in his files too?

This is what I meant when saying that basing your PRs on unmerged content is work-at-risk (I don't think you could avoid it in this case due to time constraints).

In short, you don't have to take much in considerations, but @thetaxas2468 will have to fetch your code and rebase his PR on top of it (he should not add the files manually on his own as this will generate more conflicts).

At this stage, I don't see benefits for Karam to fetch your PR branch and rebase on top of it, but instead to continue working on what he has in place right now, and do the rebase only once your PR is merged (rebase on top of upstream/main after the PR is merged).
The reason for that being is that I don't think there will be any benefit for him in taking your up-to-date code right now vs. once its merged, and it will only mean more conflict-resolution work for each time a rebase is done (it's a nice exercise but not a lot of fun...).

Bottom line, you should do your changes. Once your PR is merged, he should fetch upstream and rebase on top of upstream/main and resolve all conflicts.

**Comment by LiorSilberman** [comment]
> > 1. If I want to rename the directory, do I need to push new PR for this?
> 
> Your PR wants to introduced that directory, and the PR was not merged yet, so you need the PR content, rather than to push a new PR (or new commit). You need to:
> 
> 1. Make your modifications.
> 2. use `git status` to see the unstaged changes.
> 3. You'll see the old files removed and the new ones added.
> 4. Use `git add` to add the changes for both removed and added files.
> 5. use `git commit --amend` to amend your last commit.
> 6. use `git push` with the `--force` option to overwrite your remote branch with your local changes.

@yftacherzog 
when I try to push my changes with "git push --force", I got a fatal message: "The current branch Django has no upstream branch. To push the current branch and set the remote as upstream, use

    git push --set-upstream origin Django"

I'm not sure why it happend. can you explain me please?

**Comment by yftacherzog** [comment]
@LiorSilberman let me reply on Slack so the context will be a little clearer.

File: buy_together/wsgi.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+For more information on this file, see
+https://docs.djangoproject.com/en/4.1/howto/deployment/wsgi/
+"""
+
+import os
+
+from django.core.wsgi import get_wsgi_application
+
+os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'BuyTogether.settings')
```
Comment (yftacherzog) [inline buy_together/wsgi.py:14]: I see multiple references to `BuyTogether`, and I tend to think those refer to the files under the renamed directory. Did you test that the environment comes up after your change? It might be better to recreate it with the corrected name just to be sure.

Reminding, If you get to make fixes:
1. make the changes.
2. add them with git add (both added and deleted files).
3. git commit --amend
4. git push --force origin HEAD
